### PR TITLE
feat(presets/regex): Add preset for appVersion in helm charts

### PR DESCRIPTION
**Pull request by jkroepke**
## Changes

Adds a new regex preset for appVersion in Chart.yaml

## Context

See #13968 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/regex-managers.ts [hunk 1]
```diff
@@ -14,4 +14,17 @@ export const presets: Record<string, Preset> = {
       },
     ],
   },
+  helmChartYamlAppVersions: {
+    description: 'Update `appVersion` value in helm chart Chart.yaml.',
+    regexManagers: [
+      {
+
```
Comment (HonkingGoose) [inline lib/config/presets/internal/regex-managers.ts:21]: Please run `yarn prettier-fix`, because there's an extra newline. 😉

Comment (jkroepke) [reply lib/config/presets/internal/regex-managers.ts]: done

**Comment by HonkingGoose** [comment]
Hi @jkroepke,

Thanks you for applying my suggestion. 😄 

Next time, please don't force push to your pull request branch. Here's why: [^quote]

> ## Do not force push to your pull request branch
> 
> Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. PRs will always be squashed by us when we merge your work. Commit as many times as you need in your pull request branch.
> 
> If you are updating your PR branch from within the GitHub PR interface, then only use the default "Update branch" button.
> 
> Force pushing a PR is OK when:
> 
> - you need to make large changes on a PR which require a full review anyway
> - you need to bring the branch up-to-date with the target branch and incorporating the changes is more work than to create a new PR

[^quote]: [Renovate's contributing guide](https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#do-not-force-push-to-your-pull-request-branch)

**Comment by rarkins** [comment]
Is it possible to supplement this with an example in https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/regex/readme.md which references this preset?

**Comment by jkroepke** [comment]
> Is it possible to supplement this with an example in [`main`/lib/modules/manager/regex/readme.md](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/regex/readme.md?rgh-link-date=2022-12-15T12%3A08%3A05Z) which references this preset?

Not sure, if its worth. Added.

File: lib/config/presets/internal/regex-managers.ts [hunk 2]
```diff
@@ -14,4 +14,16 @@ export const presets: Record<string, Preset> = {
       },
     ],
   },
+  helmChartYamlAppVersions: {
+    description: 'Update `appVersion` value in helm chart Chart.yaml.',
```
Comment (HonkingGoose) [inline lib/config/presets/internal/regex-managers.ts:18]:
```suggestion
    description: 'Update `appVersion` value in Helm chart Chart.yaml.',
```

Helm is a proper noun, so we should capitalize it.

File: lib/modules/manager/regex/readme.md [hunk 3]
```diff
@@ -128,9 +128,40 @@ This is because Handlebars escapes special characters with double braces (by def
 By adding `renovate: datasource=` and `depName=` comments to the `Dockerfile` you only need _one_ `regexManager` instead of _four_.
 The `Dockerfile` is documented better as well.
 
-The syntax in the example is arbitrary and you can set your own syntax.
+The syntax in the example is arbitrary, and you can set your own syntax.
 If you do, update your `matchStrings` regex!
 
+Simplified version of such comments can be used to increase to readability.
```
Comment (HonkingGoose) [inline lib/modules/manager/regex/readme.md:134]:
```suggestion
You can use simple versions of such comments to increase readability.
```

File: lib/modules/manager/regex/readme.md [hunk 4]
```diff
@@ -128,9 +128,40 @@ This is because Handlebars escapes special characters with double braces (by def
...
+
+For example the `appVersion` property in a Chart.yaml of a helm chart is always referenced to an docker image. In such scenarios, some values
+can be hard-coded. For example:
```
Comment (HonkingGoose) [inline lib/modules/manager/regex/readme.md:137]:
```suggestion
For example the `appVersion` property in a `Chart.yaml` of a Helm chart is always referenced to an Docker image.
In such scenarios, some values can be hard-coded.
For example:
```

File: lib/modules/manager/regex/readme.md [hunk 5]
```diff
@@ -128,9 +128,40 @@ This is because Handlebars escapes special characters with double braces (by def
...
+apiVersion: v2
+name: amazon-eks-pod-identity-webhook
+description: A Kubernetes webhook for pods that need AWS IAM access
+version: 1.0.3
+type: application
+# renovate: image=amazon/amazon-eks-pod-identity-webhook
+appVersion: 'v0.4.0'
+```
+
+Using the regexManagers below, renovate will look for available docker tag of the image `amazon/amazon-eks-pod-identity-webhook`.
```
Comment (HonkingGoose) [inline lib/modules/manager/regex/readme.md:149]:
```suggestion
Using the `regexManagers` below, Renovate looks for available Docker tags of the image `amazon/amazon-eks-pod-identity-webhook`.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Comment by jkroepke** [comment]
Thanks for the reviews

File: lib/config/presets/internal/regex-managers.ts [hunk 6]
```diff
@@ -14,4 +14,16 @@ export const presets: Record<string, Preset> = {
...
     ],
   },
+  helmChartYamlAppVersions: {
+    description: 'Update `appVersion` value in Helm chart Chart.yaml.',
+    regexManagers: [
+      {
+        datasourceTemplate: 'docker',
+        fileMatch: ['(^|/)Chart\\.yaml$'],
+        matchStrings: [
+          '#\\s*renovate: image=(?<depName>.*?)\\s+appVersion:\\s*[\\"\\']?(?<currentValue>[\\w+\\.\\-]*)',
```
Comment (viceice) [inline lib/config/presets/internal/regex-managers.ts:24]:
```suggestion
          '#\\s*renovate: image=(?<depName>.*?)\\s+appVersion:\\s*["\']?(?<currentValue>[\\w+\\.\\-]*)',
```
looks like wrong escapes

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**